### ISSUE TITLE
18 tokaideo, getting error  "undefined method "sort_order_key" for <Engine::Bank> " , unable to proceed

### ISSUE THREAD

**Issue by kshgrshrm** [tobymao/18xx#9839]
**What's your Game ID?**  https://18xx.games/game/138956

**How to Reproduce the Bug?** Atomsk player has tried every action combination possible but they are not able to complete their turn by presing "Skip companies". As soon as they do they get the error mentioned in the title

<!--
What steps cause the unexpected behavior? pressing "skip companies" 

**Which Browser / Platform?** all platforms.  firefox 119.0

**Comment by benjaminxscott** [comment]
In hotseat I get the likely related error:

`comparison of `[[[1, <Engine::Bank>], 15, "Sleeper Train"], <Engine::Company:ST>] with [[[1, <Engine::Corporation:YSL>],10,"Sasago Railway Tunnel"], <Engine::Company:SRT>] failed`

https://gist.github.com/benjaminxscott/e16ded0f2b41f1bdc757579a1bf8bb0e

![Screenshot 2023-11-02 10 11 03 AM](https://github.com/tobymao/18xx/assets/1711810/625a274f-1f0f-4094-97aa-7981ab0f552a)

**Comment by andrewzwicky** [comment]
I looked into this a little bit.  In this particular game, the Sleeper Train was left behind when not claimed and remained in the Bank.  I'll write a PR to fix.

### Summary
https://18xx.games/game/138956?action=23 at this point in the game, someone buys the second to last private, no one has any money, so no one has valid actions, and the Sleeper Train private remains in the bank.  Later on in the game, the privates need to pay out and there is sorting code that tries to compare the Bank to a Corporation.

There are two issues I can see.
1. The initial stock round allows players to buy/bid on both private companies and normal corporations, which seems out of the ordinary.
2. The privates that are not claimed are not always properly closed.

The correction is probably to make sure companies are closed properly in all cases.

### Details
Fast Forward to later in the game, at the start of the first operating round where privates have been bought in by corporations.  At the start of OR2.2, the games tries to pay out companies to their respective owners.  In the code for paying out companies, there is some sorting code: https://github.com/tobymao/18xx/blob/40c49f97d2eabc50b4d642779d145d3df539be75/lib/engine/game/base.rb#L1480-L1499

Specifically, the sort_by! code.  In all of the previous iterations, the privates were completely owned by players, so even though the company was still sitting in the bank, the second part of the comparison didn't need to be evaluated.  Only when a Corporation owned a private, did the sort have to fall back to comparing `company.owner`.  Then because it's trying to compare the Bank to a Corporation, it fails.

There is code in the game logic to try to close companies that make it through the initial round, but the default is to make the initial round a normal `Engine::Round::Stock`,

https://github.com/tobymao/18xx/blob/40c49f97d2eabc50b4d642779d145d3df539be75/lib/engine/game/g_18_tokaido/game.rb#L231-L235

Basically `init_round_finished` needs to be called somewhere after the initial round is done to clear out uncollected companies.  It's currently missing in both Auction and Stock.

https://github.com/tobymao/18xx/blob/40c49f97d2eabc50b4d642779d145d3df539be75/lib/engine/game/g_18_tokaido/game.rb#L294-L317
